spec/rubocop/cop/style/redundant_array_flatten_spec.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::RedundantArrayFlatten, :config do
+  it 'registers an offense for `x.flatten.join`' do
+    expect_offense(<<~RUBY)
+      x.flatten.join
+        ^^^^^^^^^^^^ Prefer `Array#join` without `#flatten` to join nested arrays.
```

Comment:
Based on the implementation, can you make the following adjustments for offense range?

```suggestion
      x.flatten.join
        ^^^^^^^^ Remove the redundant `flatten`.
```